File: pub.sub.pattern1/Subject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace pub.sub.pattern1
{
    class Subject
    {
        public int State { get; set; }

        public event EventHandler<int> StateChange;

        public void SomeBusinessLogic() {
            Console.WriteLine("Starting state change");

            State = new Random().Next(0, 10);
            Thread.Sleep(5);

            Console.WriteLine("New State value" + State);
            OnStateChange();
        }

        public void OnStateChange()
        {
            StateChange(this, State);
        }
    }
}

File: pub.sub.vbnet/Subject.vb

Imports System.Threading

Public Class Subject
    Public Event StateChange As EventHandler(Of Integer)
    Public Property State As Integer

    Public Sub SomeBusinessLogic()
        State = New Random().Next(1, 10)
        Thread.Sleep(15)
        RaiseEvent StateChange(Me, State)
    End Sub

End Class
